great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py

```diff
@@ -140,8 +140,14 @@ def _build_parameters(
         )
 
         # As a simplification, apply reduction to scalar in case of one-dimensional metric (for convenience).
-        if reduce_scalar_metric and metric_values.shape[1] == 1:
-            metric_values = metric_values[:, 0]
+        if (
+            reduce_scalar_metric
+            and isinstance(metric_values, list)
+            and len(metric_values) == 1
+            and isinstance(metric_values[0], AttributedResolvedMetrics)
+            and metric_values[0].metric_values.shape[1] == 1
+        ):
+            metric_values = metric_values[0].metric_values[:, 0]
 
         return (
```

Comment:
Can we change the return type of this method from `Tuple(Any, dict)` to Tuple(MetricValues, MetricComputationDetails)`? Also, should `MetricComputationDetails` be an attribute of `MetricValues` so we can avoid the Tuple all together? Every time you return a Tuple, it's one more thing for the caller to worry about